string source = rogr
string target = watertank
time timestamp

